blockchain/chain_sync/src/sync.rs

```diff
@@ -851,24 +862,40 @@ mod tests {
         });
     }
 
+    fn dummy_header() -> BlockHeader {
+        BlockHeader::builder()
+            .miner_address(Address::new_id(1000).unwrap())
+            .messages(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .message_receipts(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .state_root(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .build()
+            .unwrap()
+    }
     #[test]
     fn chainsync_constructor() {
-        let db = MemoryDB::default();
+        let db = Arc::new(MemoryDB::default());
+        let mut chain_store = ChainStore::new(db);
         let (local_sender, _test_receiver) = channel(20);
         let (_event_sender, event_receiver) = channel(20);
 
+        let gen_header = dummy_header();
+        chain_store.set_genesis(gen_header.clone()).unwrap();
```

Comment:
```suggestion
        chain_store.set_genesis(gen_header).unwrap();
```
Not needed on all these, right?